### PR TITLE
support kwargs in graph/module

### PR DESCRIPTION
**Pull request by L1aoXingyu**
This PR supported key-value as arguments and return values of nn.Graph/nn.Module in LiBai.

With this feature, we can get specific loss values when there are multiple losses.

For example, we can get the following LOG on the console as a comparision. 

Key-value 

```shell
[01/24 18:56:14 lb.utils.events]:  eta: 1:52:46  iteration: 3/10000  consumed samples: 48  total_loss: 10.87  mlm loss: 10.14  sop loss: 0.7407  time: 0.6769(23.64)  data_time: 0.2903  lr: 2.00e-06  
[01/24 18:56:15 lb.utils.events]:  eta: 1:54:14  iteration: 4/10000  consumed samples: 64  total_loss: 10.89  mlm loss: 10.14  sop loss: 0.7518  time: 0.6858(23.33)  data_time: 0.2195  lr: 3.00e-06  
[01/24 18:56:15 lb.utils.events]:  eta: 1:54:38  iteration: 5/10000  consumed samples: 80  total_loss: 10.87  mlm loss: 10.13  sop loss: 0.7407  time: 0.6866(23.30)  data_time: 0.1798  lr: 4.00e-06  
[01/24 18:56:16 lb.utils.events]:  eta: 1:54:49  iteration: 6/10000  consumed samples: 96  total_loss: 10.87  mlm loss: 10.12  sop loss: 0.7309  time: 0.6876(23.27)  data_time: 0.1509  lr: 5.00e-06  
```

w/o Key-value
```shell
[01/24 19:01:31 lb.utils.events]:  eta: 3:53:15  iteration: 12/10000  consumed samples: 384  total_loss: 10.6  time: 1.4011(22.84)  data_time: 0.1355  lr: 1.10e-05  
[01/24 19:01:33 lb.utils.events]:  eta: 3:54:13  iteration: 13/10000  consumed samples: 416  total_loss: 10.6  time: 1.4025(22.82)  data_time: 0.1265  lr: 1.20e-05  
[01/24 19:01:34 lb.utils.events]:  eta: 3:54:08  iteration: 14/10000  consumed samples: 448  total_loss: 10.53  time: 1.4028(22.81)  data_time: 0.1179  lr: 1.30e-05  
[01/24 19:01:36 lb.utils.events]:  eta: 3:54:11  iteration: 15/10000  consumed samples: 480  total_loss: 10.45  time: 1.4041(22.79)  data_time: 0.1108  lr: 1.40e-05  
[01/24 19:01:37 lb.utils.events]:  eta: 3:54:09  iteration: 16/10000  consumed samples: 512  total_loss: 10.37  time: 1.4053(22.77)  data_time: 0.1043  lr: 1.50e-05  
```

PS: This PR merging depends on the following oneflow PR merging. 
https://github.com/Oneflow-Inc/oneflow/pull/7309

### CODE REVIEW

File: libai/models/bert_model.py [hunk 1]
```diff
@@ -341,29 +343,28 @@ def __init__(self, cfg):
 
     def forward(
         self,
-        input_ids,
-        attention_mask,
+        tokens,
```
Comment (dangkai4u) [inline libai/models/bert_model.py:346]: 这里为什么要改名啊，input_ids和attention_mask是bert参数比较常见的命名

File: libai/models/bert_model.py [hunk 2]
```diff
@@ -341,29 +343,28 @@ def __init__(self, cfg):
...
+        outputs = self.bert(tokens, padding_mask, tokentype_ids)
         sequence_output, pooled_output = outputs[:2]
 
         sequence_output, seq_relationship_score = self.cls(sequence_output, pooled_output)
 
         prediction_scores = self.lm_logits(sequence_output, self.bert.word_embeddings_weight())
 
         if lm_labels is not None and ns_labels is not None:
-            total_loss = self.loss_func(
+            return self.loss_func(
```
Comment (dangkai4u) [inline libai/models/bert_model.py:361]: 这里强制训练MLM和NSP任务，但有时候只训练一个，所以建议分四种情况讨论

File: libai/models/bert_model.py [hunk 3]
```diff
@@ -341,29 +343,28 @@ def __init__(self, cfg):
...
                 prediction_scores,
                 lm_labels,
                 loss_mask,
                 seq_relationship_score,
                 ns_labels,
             )
-            return total_loss
         else:
             return prediction_scores, seq_relationship_score
```
Comment (CPFLAME) [inline libai/models/bert_model.py:369]: 这里是不是也可以改成dict的输出？
不然有loss的时候返回是dict，网络前向则是返回两个变量， 不统一。

**Comment by CPFLAME** [comment]
https://github.com/Oneflow-Inc/libai/blob/9b1747492f0dde9085127867485d0e987fed71d6/libai/evaluation/evaluator.py#L167-L170
这里也得相应的改成dict的形式。 不过evaluator可以等你的这个pr合并了我再来改。

File: libai/models/bert_model.py (hunk 1 above)
Comment (L1aoXingyu) [reply libai/models/bert_model.py:346]: 这里改名是因为要和 dataset 那边的输出名称对齐
https://github.com/Oneflow-Inc/libai/blob/9b1747492f0dde9085127867485d0e987fed71d6/libai/data/datasets/bert_dataset.py#L124-L131
我去改一下 dataset 吧

File: libai/models/bert_model.py (hunk 3 above)
Comment (L1aoXingyu) [reply libai/models/bert_model.py:369]: 可以

File: libai/models/bert_model.py (hunk 2 above)
Comment (L1aoXingyu) [reply libai/models/bert_model.py:361]: 这个地方我想想怎么通过配置实现不同的 loss

**Comment by L1aoXingyu** [comment]
> https://github.com/Oneflow-Inc/libai/blob/9b1747492f0dde9085127867485d0e987fed71d6/libai/evaluation/evaluator.py#L167-L170
> 
> 
> 这里也得相应的改成dict的形式。 不过evaluator可以等你的这个pr合并了我再来改。

这个等我这边合并之后你来 fix 吧 @CPFLAME

**Comment by CPFLAME** [comment]
> > https://github.com/Oneflow-Inc/libai/blob/9b1747492f0dde9085127867485d0e987fed71d6/libai/evaluation/evaluator.py#L167-L170
> > 
> > 这里也得相应的改成dict的形式。 不过evaluator可以等你的这个pr合并了我再来改。
> 
> 这个等我这边合并之后你来 fix 吧 @CPFLAME

可以的

**Comment by L1aoXingyu** [comment]
在 nn.Graph build 支持 kwargs 作为输入之后，发现几乎所有模型都可以沿用同一套 build 方法，所以去掉了 `BertForPretrainingGraph` 中的 build 函数。

之前在做 graph 模型的过程中，一直想把 nn.Graph 的构建埋在底下不让用户触碰，因为 Module -> Graph 只是动态到静态的转换，所有的构图流程都基于 Module 中的 forward 方法，在支持 kwargs 传参之后，发现这时 Graph 就只剩下 set_pipeline_stage_id 这个方法了，而且每个模型都不一样，需要定制。

然而当用户不需要跑流水并行的时候，其实并不用设置这个方法，所以决定将设置 stage id 和 Module 绑定，作为 Module 的 staticmethod， 用户也可以不设置，如果设置的话，将按照用户配置的方式设置流水并行的 stage id。

File: libai/models/bert_model.py [hunk 4]
```diff
@@ -222,11 +204,14 @@ def forward(self, lm_output, lm_labels, loss_mask, binary_logits, ns_labels):
...
-        ).mean()
-        losses = masked_lm_loss + sop_loss
-        return losses
+        loss_dict = {"lm loss": masked_lm_loss}
+
+        if self.add_binary_head:
+            sop_loss = flow._C.cross_entropy(
+                binary_logits, ns_labels, ignore_index=-1, reduction="none"
+            ).mean()
+            loss_dict["sop loss"] = sop_loss
```
Comment (strint) [inline libai/models/bert_model.py:213]: key的命名，是不是采用sop_loss这样比较标准

Comment (L1aoXingyu) [reply libai/models/bert_model.py:213]: 之前我看到这里是这样写的，所以就沿用了

https://github.com/NVIDIA/Megatron-LM/blob/9a8b89acd8f6ba096860170d0e30ddc0bc2bacd4/pretrain_bert.py#L91-L92

等会儿我改一下，统一加 `_`

**Review by rentainhe — APPROVED** [review]
